torchft/checkpointing/pg_transport.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import pickle
+import time
+from contextlib import contextmanager
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Generator, List, Tuple, TypeVar, cast
+
+import torch
+from torch.distributed import Work
+from torch.distributed.tensor import DTensor
+from torch.utils._pytree import tree_flatten, tree_unflatten
+
+from torchft.checkpointing.transport import CheckpointTransport
+from torchft.process_group import ProcessGroup
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+@dataclass
+class _TensorMeta:
+    shape: torch.Size
+    dtype: torch.dtype
+    storage_offset: int
+    stride: Tuple[int, ...]
+    nbytes: int
+
+
+@dataclass
+class _DTensorMeta:
+    local: _TensorMeta
+    spec: object
+
+
+@dataclass
+class _StateDictMeta:
```

Comment:
I was thinking the `_StateDictMeta` and dataclasses could probably use some more comments since that's pretty important in determining how we serialize / deserialize and being able to update them. I am kinda curious how DCP handles this metadata when it transfers and if we have existing infra we can use? @fegin do you know?

Reply:
Added comments and updated field names to make it clearer

Reply:
@H-Huang similar to what is does here. Some logic is the same and we can see if we want to unify the code in the future.